lib/domain/usecase/search/pre_search_recent_contacts_interactor.dart

```diff
@@ -0,0 +1,48 @@
+import 'package:dartz/dartz.dart';
+import 'package:fluffychat/app_state/failure.dart';
+import 'package:fluffychat/app_state/success.dart';
+import 'package:fluffychat/domain/app_state/search/pre_search_state.dart';
+import 'package:matrix/matrix.dart';
+
+class PreSearchRecentContactsInteractor {
+  PreSearchRecentContactsInteractor();
+
+  Stream<Either<Failure, Success>> execute({
+    required List<Room> recentRooms,
+    int? limit,
+  }) async* {
+    try {        
+      final List<User> result = [];
+
+      for (final room in recentRooms) {
+        final users = room.getParticipants()
+          .where((user) => user.membership.isInvite == true && user.displayName != null)
+          .toSet();
+
+        for (final user in users) {
+          final isDuplicateUser = result
+            .any((existingUser) => existingUser.displayName == user.displayName);
```

Comment:
IMO, you should check duplicate user using matrixId, because displayName maybe different in many client app

Reply:
done, using `id`